File: solutions/clojure/word-count/1/src/word_count.clj
(ns word-count)
(defn freq [c] (reduce (fn [m e] (assoc m e (inc (m e 0)))) {} c))

(defn word-count [s] 
    (let [s (.toLowerCase  s)
          s (.replaceAll s "[^a-z0-9 ]" "")
          words (map (partial apply str) (.split s " "))
          words (remove #{""} words)]
    (freq words)))

File: solutions/python/word-count/1/word_count.py
def word_count(phrase):
    phrase2 = "" 
    badcharacters = set(",:_!!&@$%^&.")
    for ch in phrase.lower():
       if (ch in badcharacters):
           phrase2 = phrase2 + " " 
       else:
           phrase2 = phrase2 + ch
    words = phrase2.split() 
    wordmap = {}
    for word in words:
       word = word.strip("'")
       if (not word in wordmap):
           wordmap[word] = 1
       else:
           wordmap[word] = wordmap[word] + 1
    return wordmap 
